src/wp-includes/class-wp-block-type.php

```diff
@@ -213,6 +213,18 @@ class WP_Block_Type {
 	 */
 	public $style_handles = array();
 
+	/**
+	 * Block hooks for this block type.
+	 *
+	 * A block hook is specified by a block type and a relative position.
+	 * The hooked block will be automatically inserted in the given position
+	 * next to the "anchor" block whenever the latter is encountered.
+	 *
+	 * @since 6.4.0
+	 * @var array|null
```

Comment:
What would be the use case for `null` here?

Looking at the usage, there is an assumption that it's actually an array:
```php
foreach ( $block_type->block_hooks as $anchor_block_type => $relative_position ) { 
```

Reply:
Yeah, good point. TBH I was mostly unsure about how default values are handled in block registration. Anyway, defaulting to empty array and never even allowing `null` makes sense to me 👍

Reply:
0267cc936c